### PR TITLE
quilt: enable strictDeps

### PR DESCRIPTION
**Pull request by FliegendeWurst**
ref. #178468

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/qu/quilt/package.nix [hunk 1]
```diff
@@ -42,6 +42,14 @@ stdenv.mkDerivation rec {
     unixtools.getopt
   ];
 
+  strictDeps = true;
+
+  configureFlags = [
+    # configure only looks in $PATH by default,
+    # which does not include buildInputs if strictDeps is true
+    "--with-perl=${lib.getExe perl}"
+  ];
```
Comment (superherointj) [inline pkgs/by-name/qu/quilt/package.nix:51]:
```suggestion
```
I suggest you add `perl` to `nativeBuildInputs`.

**Comment by emilazy** [comment]
Commit message needs changing too if you’re going with that suggestion, I suppose?

**Comment by superherointj** [comment]
1) Are you sure `perl` is not necessary at runtime? My bias would be to keep perl at `buildInputs` just in case. (I don't have good context on this.)

OPTIONAL: (Only if you want to do extra changes. I don't really care about these, only pointing out for the sake of being comprehensive. You can ignore this part if you wish. No problem!)

2) shell completion exist but is not packaged.
  https://git.savannah.nongnu.org/cgit/quilt.git/tree/bash_completion

MUCH LESS MEANINGFUL (no practical effect):

3) `hash` is favored over `sha256`.

4) nixfmt

5) remove `with lib;`

6) meta.description should be first.

**Comment by FliegendeWurst** [comment]
About (1), that's what I did originally. Actually, I will check if it is used at build time. It definitely is at runtime.

(3) to (6) I will leave for treewide fixes.

**Comment by superherointj** [comment]
> About (1), that's what I did originally. Actually, I will check if it is used at build time. It definitely is at runtime.

The dependency should have just been repeated at `buildInput` and `nativeBuildInput` in this case.

**Comment by superherointj** [comment]
> (3) to (6) I will leave for treewide fixes.

No problem.

**Comment by FliegendeWurst** [comment]
I built both quilt and `pkgsCross.aarch64-multiplatform.quilt`.

**Comment by superherointj** [comment]
I dont think configure flag is needed. Only had to repeat perl in nativeBuildInputs.

**Comment by FliegendeWurst** [comment]
No, perl is not used at build time. Otherwise the cross build would have failed, right?

**Comment by superherointj** [comment]
> No, perl is not used at build time. Otherwise the cross build would have failed, right?

Even if you remove that configure flag?

File: pkgs/by-name/qu/quilt/package.nix (hunk 1 above)
Comment (superherointj) [inline pkgs/by-name/qu/quilt/package.nix:51]:
```suggestion
  nativeBuildInputs = [ perl ];
```
I expect this to work as long `perl` isn't removed from buildInputs.

**Comment by FliegendeWurst** [comment]
Both the regular and the cross builds fail without the flag.

File: pkgs/by-name/qu/quilt/package.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/by-name/qu/quilt/package.nix:51]: And even if it worked, the configure script would pick up the wrong Perl for the build platform.

Comment (superherointj) [reply pkgs/by-name/qu/quilt/package.nix:51]: Allright. Understood the issue. That's a runtime perl. I'm sorry for the trouble.

**Review by superherointj — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**